src/lib/app-provider/hooks/useAttachFunds.ts

```diff
@@ -0,0 +1,42 @@
+import type { Coin } from "@cosmjs/stargate";
+import { useCallback } from "react";
+
+import { AttachFundsType } from "lib/components/fund/types";
+import { useAssetInfos } from "lib/services/assetService";
+import { fabricateFunds, sortDenoms } from "lib/utils";
+
+export const useAttachFunds = () => {
+  const { assetInfos } = useAssetInfos();
+
+  return useCallback(
+    (
+      attachFundsOption: AttachFundsType,
+      assetsJsonStr: string,
+      assetsSelect: Coin[]
```

Comment:
I prefer naming this selectedAssets

Reply:
the assetsSelect represent the asset in the Select asset and fill amount section. selectedAssets might cause a confusion too